### PR TITLE
Modify insert operation's behavior

### PR DESCRIPTION
**Pull request by morningman**
Before changing default insert operation to streaming load, even if the insert operation
encounter errors, a label will be returned to the user, and user
can use this label to check the insert load job's status.

After changing the insert operation, if the the insert operation encounter errors, 
an exception will be thrown to user client directly without any label.

This new usage pattern is not friendly to the already existed users, which is forcing
them to change their way of using insert operation.

So I add a new FE config 'using_old_load_usage_pattern', default is false.
If set to true, a label will be returned to user even if the insert operation encounter errors.

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/load/loadv2/InsertLoadJob.java [hunk 1]
```diff
@@ -46,14 +50,20 @@ public InsertLoadJob() {
...
         this.loadStartTimestamp = createTimestamp;
         this.finishTimestamp = System.currentTimeMillis();
-        this.state = JobState.FINISHED;
-        this.progress = 100;
+        if (Strings.isNullOrEmpty(failMsg)) {
+            this.state = JobState.FINISHED;
+            this.progress = 100;
+        } else {
+            this.state = JobState.CANCELLED;
+            this.failMsg = new FailMsg(CancelType.ETL_RUN_FAIL, failMsg);
```
Comment (EmmyMiao87) [inline fe/src/main/java/org/apache/doris/load/loadv2/InsertLoadJob.java]: There is no ETL_RUN_FAIL in streaming load. Please use LOAD_RUN_FAIL instead.

File: fe/src/main/java/org/apache/doris/transaction/TransactionCommitFailedException.java [hunk 2]
```diff
@@ -22,6 +22,8 @@
     
     private static final long serialVersionUID = -2528170792631761535L;
 
+    public static final String NO_DATA_TO_LOAD_MSG = "all partitions have no load data";
```
Comment (EmmyMiao87) [inline fe/src/main/java/org/apache/doris/transaction/TransactionCommitFailedException.java]: You don't need to judge NO_DATA_TO_LOAD_MSG. In my option, all of exceptions need to be catch and record in old user pattern

File: fe/src/main/java/org/apache/doris/load/loadv2/InsertLoadJob.java (hunk 1 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/load/loadv2/InsertLoadJob.java]: OK

File: fe/src/main/java/org/apache/doris/transaction/TransactionCommitFailedException.java (hunk 2 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/transaction/TransactionCommitFailedException.java]: OK

**Review by imay — APPROVED** [review]

**Pull request merged**